src/Controller/Plugin/Forward.php

```diff
@@ -179,32 +179,39 @@ protected function detachProblemListeners(SharedEvents $sharedEvents)
             foreach ($eventArray as $eventName => $classArray) {
                 $results[$id][$eventName] = [];
                 $events = $this->getSharedListenersById($id, $eventName, $sharedEvents);
-                foreach ($events as $priority => $currentEvent) {
-                    $currentCallback = $currentEvent;
-
-                    // zend-eventmanager v2 compatibility:
-                    if ($currentCallback instanceof CallbackHandler) {
-                        $currentCallback = $currentEvent->getCallback();
-                        $priority        = $currentEvent->getMetadatum('priority');
+                foreach ($events as $priority => $currentPriorityEvents) {
+                    // v2 fix
+                    if (!is_array($currentPriorityEvents)) {
+                        $currentPriorityEvents = [$currentPriorityEvents];
                     }
+                    // v3
```

Comment:
This comment can be removed, as the logic below executes for either version. (I'll do that during merge.)